crates/xtask-bump-check/src/xtask.rs

```diff
@@ -361,7 +362,7 @@ fn check_crates_io<'a>(
 ) -> CargoResult<()> {
     let source_id = SourceId::crates_io(config)?;
     let mut registry = PackageRegistry::new(config)?;
-    let _lock = config.acquire_package_cache_lock()?;
+    let _lock = config.acquire_package_cache_lock(CacheLockMode::DownloadExclusive)?;
```

Comment:
Could this be split out into its own PR?  I feel like there is a lot to unpack with this
- Clearly documenting interactions with old cargo
- Working out how callers are supposed to know which lock type to use, especially when copy/pasting
- Strategies for correctness

Reply:
Sure, I have posted #12706 to separate out that change.  I can separate out others as desired as well, just let me know.